### DISCUSSION TITLE
Checking if folder exists works weird

### DISCUSSION THREAD

**Discussion by cshyshtof** [SilentVoid13/Templater#587] [Help] [answered]
I use Obsidian for project management and I have separate folders for each project in a form of "Projects/Project ID". I use templater to create new meeting notes for each project. I want all meeting to be stored in Meetings directory. Now, when I create the first meeting, the folder is created properly ""Projects/Project ID/Meetings", which means, all paths are correct. However, when I create another meeting, I get the error with message "Folder already exists", which means, checking if folder exists does not work as it tries to create the folder again. I use the template while being in project's directory. Any clue? Snippet:

```js
destDir = 'Projects/' + tp.file.folder() + '/Meetings';
fileName = tp.date.now("YYYY-MM-DD");
destFile = destDir + '/' + fileName;

await tp.file.rename(fileName);

if (! tp.file.exists(destDir)) {
   await this.app.vault.createFolder(destDir);
}

await tp.file.move(destFile);
```

**Comment by drmikecrowe** [comment]
I think it needs an `await`

**Reply by cshyshtof** [reply]
But all file-based operations already have `await`. Adding `await tp.file.exists` changes nothing.

**Reply by drmikecrowe** [reply]
whoops -- i was wrong.  Ignore my comment

**Comment by liamcain** [comment, marked as answer]
`tp.file.exists` checks for the existence of files only (not folders). If you want to check if the folder exists, you'll need to use Obsidian's `vault.exists()` function. Note: this function is async so you'll need to call it with `await`.

Here's an example of a template that moves a file to a folder that may or may not exist yet:

```
<%*
if (!(await this.app.vault.exists('a/b'))) {
  await this.app.vault.createFolder('a/b');
}
await tp.file.move('a/b/c');
%>
```

**Comment by cshyshtof** [comment]
It works, thanks :-) Many samples on the Internet used `tp.file.exists` for checking directory, but appearently they were wrong...
